# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1258538805 -3600
#      Wed Nov 18 11:06:45 2009 +0100
# Node ID 6631c61a4a4e13d926b9b919f9fea72630e48d8b
# Parent  9b62f2d8de6dfcdb689e3e7d32faa9cbb39a310c
fix typos in previous change

diff --git a/src/ov-flt-re-mat.cc b/src/ov-flt-re-mat.cc
--- a/src/ov-flt-re-mat.cc
+++ b/src/ov-flt-re-mat.cc
@@ -704,17 +704,17 @@ do_rc_map (const FloatNDArray& a, FloatC
             rc(j) = rr(j);
 
           rc(i) = tmp;
 
           for (octave_idx_type j = i+1; j < n; j++)
             {
               OCTAVE_QUIT;
 
-              rc(j) = fcn (a(i));
+              rc(j) = fcn (a(j));
             }
 
           return new octave_float_complex_matrix (rc);
         }
     }
 
   return rr;
 }
diff --git a/src/ov-re-mat.cc b/src/ov-re-mat.cc
--- a/src/ov-re-mat.cc
+++ b/src/ov-re-mat.cc
@@ -732,17 +732,17 @@ do_rc_map (const NDArray& a, Complex (&f
             rc(j) = rr(j);
 
           rc(i) = tmp;
 
           for (octave_idx_type j = i+1; j < n; j++)
             {
               OCTAVE_QUIT;
 
-              rc(j) = fcn (a(i));
+              rc(j) = fcn (a(j));
             }
 
           return new octave_complex_matrix (rc);
         }
     }
 
   return rr;
 }
